common/core/nesteddisplay.cpp

```diff
@@ -67,10 +133,109 @@ bool NestedDisplay::SetPowerMode(uint32_t /*power_mode*/) {
   return true;
 }
 
-bool NestedDisplay::Present(std::vector<HwcLayer *> & /*source_layers*/,
+bool NestedDisplay::Present(std::vector<HwcLayer *> &source_layers,
                             int32_t * /*retire_fence*/,
                             bool /*handle_constraints*/) {
-  // TODO implement Present
+
+  int ret = 0;
+  size_t size = source_layers.size();
+  const uint32_t* pitches;
+  const uint32_t* offsets;
+  HWCNativeHandle sf_handle;
+  size_t buffer_number = 0;
+  size_t info_size = sizeof(vm_buffer_info);
+  size_t header_size = sizeof(vm_header);
+
+  for (size_t layer_index = 0; layer_index < size; layer_index++) {
+    HwcLayer* layer = source_layers.at(layer_index);
+    if (!layer->IsVisible())
+      continue;
+
+    HwcRect<int> display_frame = layer->GetDisplayFrame();
+    sf_handle = layer->GetNativeHandle();
+    auto search = mHyperDmaExportedBuffers.find(sf_handle);
+    if(search == mHyperDmaExportedBuffers.end()){
+      DrmBuffer* buffer = new DrmBuffer();
+      buffer->InitializeFromNativeHandle(sf_handle, buffer_handler_);
+
+      if(mHyperDmaBuf_Fd > 0 && buffer->GetPrimeFD()> 0){
+        struct ioctl_hyper_dmabuf_export_remote msg;
+        memset(&msg,0,sizeof(ioctl_hyper_dmabuf_export_remote));
+
+        /* TODO: add more flexibility here, instead of hardcoded domain 0*/
+        msg.remote_domain = 0;
+        msg.dmabuf_fd = buffer->GetPrimeFD();
+
+        ret = ioctl(mHyperDmaBuf_Fd, IOCTL_HYPER_DMABUF_EXPORT_REMOTE, &msg);
```

Comment:
This needs to be under Nesteddisplay compiler checks.

Reply:
I don't think there need compiler check, if no NESTED_DISPLAY_SUPPORT, this display function Present will never be called.

Reply:
sure, will follow it.